pkg/operator/apis/monitoring/v1/pod_types.go

```diff
@@ -250,16 +262,26 @@ func (pm *PodMonitoring) endpointScrapeConfig(index int, projectID, location, cl
 	}
 
 	return endpointScrapeConfig(
-		pm.GetKey(),
+		pm,
```

Comment:
thought (non-blocking): this is a bit of a code-smell

We're running in an instance method, calling a pure helper function, and passing the instance to it. This feel like it should just be flattened in-line to avoid the confusion.

Maybe part of this stems from the fact that we're passing in a mutable argument at the top-level with the `pool PrometheusSecretConfigs` object.

This works - so I'm not here to block it being merged, but I do have mild concerns about long-term maintainability of this for others without context.

Reply:
Yeah - I can see how we got here.

Addressing this would ultimately be a larger refactoring, which isn't necessary or appropriate in this PR. Thanks for explaining!